beacon-chain/rpc/beacon/blocks.go

```diff
@@ -192,6 +194,24 @@ func (bs *Server) StreamBlocks(_ *ptypes.Empty, stream ethpb.BeaconChain_StreamB
 					// One nil block shouldn't stop the stream.
 					continue
 				}
+				if featureconfig.Get().NewStateMgmt {
+					hasStateSummaryDB := bs.BeaconDB.HasStateSummary(bs.Ctx, bytesutil.ToBytes32(data.SignedBlock.Block.ParentRoot))
+					hasStateSummaryCache := bs.StateSummaryCache.Has(bytesutil.ToBytes32(data.SignedBlock.Block.ParentRoot))
+					if !hasStateSummaryDB && !hasStateSummaryCache {
+						log.WithField("blockSlot", data.SignedBlock.Block.Slot).Warn("No access to parent state to verify block signature")
+						continue
+					}
+					parentState, err := bs.StateGen.StateByRoot(bs.Ctx, bytesutil.ToBytes32(data.SignedBlock.Block.ParentRoot))
```

Comment:
Have you checked how expensive it is to look up the parent root? This also assumes we have the parent root for every block we get, is there a reason we can't use head state?

Reply:
terence suggested it will be less accurate when i asked. worth checking again